src/tests/ftest/util/ssh_connection.py

```diff
@@ -0,0 +1,137 @@
+            debug  : To print the command on console.
+        return:
+            None
+        """
+        self.address = address
+        self.timeout = timeout
+        self.debug = debug
+        self.client = client.SSHClient()
+        self.client.set_missing_host_key_policy(client.AutoAddPolicy())
+
+    def connect(self):
+        """
+        Connect the client via ssh connection
+        """
+        self.client.connect(self.address)
+
+        if self.debug:
+            print("SSH Connection open to {0}.".format(self.address))
+
+    def check_output(self, command):
+        """
+        Execute command on remote machine.
+        Args:
+            command: Command to run on remote machine.
+        return:
+            stdout: stdout of the return code
+        Raises:
+            general: if it's failed in ssh channel setup, exec command, close()
+            general: If command exit status is not 0.
+        """
+        try:
+            transport = self.client.get_transport()
+            transport.set_keepalive(20)
+            channel = transport.open_session()
+            channel.settimeout(self.timeout)
+            channel.exec_command(command)
+            stdout = channel.makefile('rb').read()
+            exit_status = channel.recv_exit_status()
+            channel.close()
+        except:
+            print("Exception in call for command {0}".format(command))
+            raise
+
+        if self.debug:
+            stderr = channel.makefile_stderr('rb').read()
+            print("[{0}] Command :{1}\nRC:{2} \nstdout : {3}\nstderr :{4}"
+                  .format(self.address,
+                          command,
+                          exit_status,
+                          stdout,
+                          stderr))
+
+        if exit_status != 0:
+            raise Exception("[{0}] Command :{1}\nRC:{2} \nstdout : {3}\n" \
+                            "stderr :{4}".format(self.address,
+                                                 command,
+                                                 exit_status,
+                                                 stdout,
+                                                 stderr))
+
+        return stdout
+
+    def call(self, command):
+        """
+        Execute command on remote machine.
+        Args:
+            command: Command to run on remote machine.
+        return:
+            exit_status: int value of exit status
+        Raises:
+            general: if it's failed in ssh channel setup, exec command, close()
+        """
+        try:
+            transport = self.client.get_transport()
+            transport.set_keepalive(20)
+            channel = transport.open_session()
+            channel.settimeout(self.timeout)
+            channel.exec_command(command)
+            exit_status = channel.recv_exit_status()
+            channel.close()
+        except:
+            print("Exception in call for command {0}".format(command))
+            raise
```

Comment:
Ideally we should raise a known exception type such that consumers of this function can write expectant try/excepts

Reply:
Same answer as mention above.